mlflow/server/js/src/components/MetricsPlotPanel.js

```diff
@@ -109,37 +114,57 @@ export class MetricsPlotPanel extends React.Component {
     return metrics;
   };
 
+  /**
+   * Handle changes in the scale type of the y-axis
+   * @param yAxisLogScale: Boolean - if true, y-axis should be converted to log scale, and if false,
+   * y-axis scale should be converted to a linear scale.
+   */
   handleYAxisLogScaleChange = (yAxisLogScale) => {
     const newLayout = _.cloneDeep(this.state.layout);
-    // If yaxis was already explicitly specified, convert range to appropriate coordinates
-    // for log axis (base 10), and vice versa. When converting to log scale, handle negative values
-    // by deferring to plotly autorange
     const newAxisType = yAxisLogScale ? "log" : "linear";
 
-    // If plot previously had no y axis range configured, simply set the axis type to log or
-    // linear scale appropriately
+    // lastLinearYAxisRange contains the last range used for a linear-scale y-axis
+    // We reset this state attribute each time the y-axis log scale changes, unless we're explicitly
+    // converting from a linear axis with negative bounds to a log scale axis.
+    let lastLinearYAxisRange = [];
+
+    // Handle special case of a linear y-axis scale with negative values converted to log scale &
+    // now being restored to linear scale, by restoring the old linear-axis range from
+    // this.state.linearYAxisRange
+    if (!yAxisLogScale && this.state.lastLinearYAxisRange &&
+        this.state.lastLinearYAxisRange.length > 0) {
```

Comment:
How do you know that the linear y-axis scale we're changing to has negative values?

Reply:
I guess here we're assuming that if `lastLinearYAxisRange` is non-empty, it contains a linear y axis range with negative values. Maybe I can update the name/add a comment to make this clearer...

Reply:
Ah got it. Your comment below really clears this up --thanks! (` We set this state attribute if and only if we're converting from ...`)